compiler/src/iree/compiler/DispatchCreation/ElementwiseOpFusion.cpp

```diff
@@ -135,24 +134,35 @@ void ElementwiseOpFusionPass::runOnOperation() {
         return areFusableAsElementwiseOps(context, fusedOperand,
                                           fuseMultiReduction);
       };
-  linalg::populateElementwiseOpsFusionPatterns(fusionPatterns,
+
+  RewritePatternSet linalgFusionPatterns(context);
```

Comment:
I feel like this change shouldn't require any changes to ElementwiseOpFusionPass and transpose on the input should be fused with the producer during transpose propagation. It just seems a bit weird that there is this implicit connection between the passes where one introduces the transpose and the other makes sure to fuse it in the optimal way.

That might require some serious changes to transpose propagation that probably aren't in the scope of this PR, so this makes sense for now. Also, `populateFuseLinalgExtOpsWithTransposes` is a bit hacky so it makes sense to run it after everything.

Reply:
@qedawkins I think you were the one mainly working on that pass. Is there a reason why we don't fuse transposes with elementwise ops during that pass?

Reply:
Yeah agreed, ideally Transpose Propagation takes care of everything, but currently for some reason the transposeOp get sunk back down. i.e if I do `test-bubbling-only` it works as expected. That's why we need the change to `populateFuseLinalgExtOpsWithTransposes` to guarantee the fuse into producer.